src/main/java/com/systemmeltdown/robotlog/outputs/ZipFileUtils.java

```diff
@@ -0,0 +1,163 @@
+  public static final String ZIP_EXTENSION = ".zip";
+
+  public static ZipOutputStream initZipFile(String dirPath, String fileName, int compressionLevel, String internalFileName)
+    throws IOException, FileNotFoundException
+  {
+    ZipOutputStream zipOut = createZipOutputStream(dirPath, fileName);
+    zipOut.setLevel(compressionLevel);
+    zipOut.putNextEntry(new ZipEntry(internalFileName));
+    return zipOut;
+  }
+
+  public static void completeZipFile(ZipOutputStream zipOut) throws IOException {
+    zipOut.closeEntry();
+    zipOut.close();
+  }
+
+  public static ZipOutputStream createZipOutputStream(String dirPath, String fileName) throws FileNotFoundException {
+    File dir = getOrCreateDir(dirPath);
+    File zipFile = getFile(dir, fileName);
+
+    System.out.println("ZipFileOutput: Creating zip file '" + zipFile + "'");
+    FileOutputStream fileOut = new FileOutputStream(zipFile);
+    BufferedOutputStream bufferedOut = new BufferedOutputStream(fileOut);
+    return new ZipOutputStream(bufferedOut);
+  }
+
+  public static File getFile(File dir, String fileName) {
+    return getFile(dir, fileName, 0);
+  }
+
+  public static File getFile(File dir, String fileName, int count) {
+    String fullFileName = fileName + (count > 0 ? "." + count : "") + ZIP_EXTENSION;
+
+    File file = new File(dir, fullFileName);
+
+    if (file.exists()) {
+      return getFile(dir, fileName, count + 1);
+    }
+    return file;
+  }
+
+  public static File getOrCreateDir(String path) {
+    File dir = new File(path);
+
+    if (!dir.exists()) {
+      String parentPath = dir.getParent();
+
+      if (parentPath == null) {
+        System.err.println("ZipFileOutput: Cannot create dir (" + path + ") for unknown parent dir");
+        return null;
+      }
+
+      File parentDir = getOrCreateDir(parentPath);
+      if (parentDir == null) {
+        // No need to log to stderr here, the parent recursive call should have already logged.
+        return null;
+      }
+
+      System.out.println("ZipFileOutput: Creating directory '" + path + "'");
+      if (!dir.mkdir()) {
+        System.err.println("ZipFileOutput: Failed to create parent directory: " + parentPath);
+        return null;
+      }
+    }
+
+    if (!dir.isDirectory()) {
+      System.err.println("ZipFileOutput: Path is not directory: " + path);
+      return null;
+    }
+    if (!dir.canWrite() || !dir.canExecute()) {
+      System.err.println("ZipFileOutput: Insufficient privileges for directory: " + path);
+      return null;
+    }
+
+    // Everything checks out.
+    return dir;
+  }
+
+  public static String printHeader(Map<String, Object> header) {
+    StringBuffer sb = new StringBuffer();
+    boolean needsComma = false;
+
+    sb.append("\"header\": { ");
+
+    for (Map.Entry<String, Object> entry : header.entrySet()) {
+      String keyStr = "\"" + entry.getKey() + "\"";
+      String valueStr = "\"" + printValue(entry.getValue(), entry.getValue().getClass()) + "\"";
```

Comment:
I think printValue should quote strings instead of quoting everything here. Any reason not to output numbers as numbers?

Reply:
Agreed, I moved the quotes to `printValue` for strings only. Thanks!